### PR TITLE
Module start and end times

### PR DESCRIPTION
**Pull request by mjk655**
Start and stop times are added to TimeModule class. In JetScapeModuleBase, tasks which are executed on a time-step basis will only run when current module time (from main clock or module clock if applicable) is within the valid time range (using IsValidModuleTime() function). Default time ranges are from 0 to 100. Different times can be set per module by module->SetTimeRange(t1,t2). An example is added to PythiaBrickTest as ClockTest.

### CODE REVIEW

File: examples/custom_examples/PythiaBrickTest.cc [hunk 1]
```diff
@@ -149,8 +168,25 @@ int main(int argc, char** argv)
   jetscape->SetXMLUserFileName("../config/jetscape_user_test.xml");
   jetscape->SetId("primary");
   jetscape->AddMainClock(mClock);
+  jetscape->AddModuleClock(mModuleClock);
```
Comment (jhputschke) [inline examples/custom_examples/PythiaBrickTest.cc:171]: Hmm, the main module jetscape should only have the main clock, not a separate module clock ...
Is this necessary? Maybe I am missing something

File: examples/custom_examples/PythiaBrickTest.cc [hunk 2]
```diff
@@ -122,7 +141,7 @@ int main(int argc, char** argv)
   //mClock->SetTimeRefFrameId("SpaceTime");
 
   // clocks here are defaulted for testing, clocks can costumized via inhererting from the MainClock/ModuleClock base classes ...
-  auto mClock = make_shared<MainClock>("SpaceTime",0,5,0.1); // JP: make consistent with reading from XML in init phase ...
+  auto mClock = make_shared<MainClock>("SpaceTime",1,5,0.1); // JP: make consistent with reading from XML in init phase ...
```
Comment (jhputschke) [inline examples/custom_examples/PythiaBrickTest.cc:144]: Maybe we can setup a test with an explicit negative start time and one test module allowing the other at t greater then t0 and see if the default catches the default start times > 0 ...

**Comment by jhputschke** [comment]
Besides the comments above looks fine. Joern

File: examples/custom_examples/PythiaBrickTest.cc (hunk 2 above)
Comment (jhputschke) [reply examples/custom_examples/PythiaBrickTest.cc:144]: for the remains modules ...

Comment (mjk655) [reply examples/custom_examples/PythiaBrickTest.cc:144]: Hi Joern, I've corrected the changes you mention above (I was doing some tests and didn't change things back). For the test with the negative times, I set it up in the current pythia brick test code and it all seems to work fine. However, are you thinking of a explicit custom example to do only that for testing purposes? If so I can add that as a separate example

Comment (jhputschke) [reply examples/custom_examples/PythiaBrickTest.cc:144]: No, in the Pythia brick test is fine for now. Make a new commit with the changes and then we can close the pull request.

**Pull request merged**